scss/components/button.scss

```diff
@@ -76,7 +76,10 @@ a.#{$block} {
       }
       &:empty {
         padding: 0;
-        &::before,
+        &::before {
+          margin-right: 0;
+          margin-left: 0;
+        }
```

Comment:
I _think_ there's a much easier way of doing this.  Instead of resetting margins back to 0, we should just _only_ add the margins when it's a button with an icon _AND_ text (not:empty).  That said, that will probably come as a future refactoring effort because it will require base class selectors (_e.g._ `.sap-icon`).  This will probably become part of #138.

For this PR, you could simply combine the `before` and `after` pseudo elements into a better sass syntax:
```scss
&:empty {
    padding: 0;

    &::before,
    &::after {
        margin-right: 0;
        margin-left: 0;
    }
}
```